tests/end-to-end/hnc/test-hnc.bats

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bats
+
+# bats file_tags=hnc
+
+setup_file() {
+  export BATS_NO_PARALLELIZE_WITHIN_FILE=true
+  export CK8S_AUTO_APPROVE="true"
+}
+
+setup() {
+  load "../../bats.lib.bash"
+  load_assert
+  with_namespace staging
+}
+
+@test "hnc dev can create subnamespace" {
+  with_test_kubeconfig wc static-dev
+  echo "# If the test stops here go to http://localhost:8000 and log in with static email user dev@example.com" >&3
+  run kubectl auth whoami
+  assert_output --partial "dev@example.com"
+  run kubectl apply -f - <<EOF
+apiVersion: hnc.x-k8s.io/v1alpha2
+kind: SubnamespaceAnchor
+metadata:
+  name: "${NAMESPACE}-tests-end-to-end"
+  namespace: "${NAMESPACE}"
+EOF
+
+  assert_success
+}
+
+@test "hnc subnamespace can create namespace" {
+  with_test_kubeconfig wc static-dev
+  echo "# If the test stops here go to http://localhost:8000 and log in with static email user dev@example.com" >&3
+  run kubectl auth whoami
+  assert_output --partial "dev@example.com"
+  run kubectl get subns "${NAMESPACE}"-tests-end-to-end --namespace "${NAMESPACE}"
```

Comment:
Question: Will this not just check that it successfully created the subnamespace? Shouldn't this be

```suggestion
  run kubectl get ns "${NAMESPACE}"-tests-end-to-end
```

Or rename it "hnc subnamespace can create subnamespaceanchor"